src/core/server/integration_tests/http/router.test.ts

```diff
@@ -561,12 +561,25 @@ describe('Cache-Control', () => {
 });
 
 describe('Handler', () => {
-  it("Doesn't expose error details if handler throws", async () => {
+  it("Doesn't expose sensitive nor uninteresting error details if handler throws", async () => {
     const { server: innerServer, createRouter } = await server.setup(setupDeps);
     const router = createRouter('/');
 
     router.get({ path: '/', validate: false }, (context, req, res) => {
-      throw new Error('unexpected error');
+      // simulate the enriched error that we get for an actual request
+      throw Object.assign(new Error('unexpected error'), {
+        meta: {
+          request: {
+            options: {
+              headers: {
+                'user-agent': 'Kibana/8.11.0',
+                authorization: 'Bearer 5eaGBBijbDxx someToken RfK7IcNAkAAAA=',
+                cookie: '5eaGBBijbDxx someCookie RfK7IcNAkAAAA=',
+              },
+            },
+          },
+        },
+      });
```

Comment:
The doc I'm looking at has `error.meta.headers` + `error.meta.request.options.headers` (covered by this test) + `error.meta.request.params.headers` 🤯 

We might want to confirm that we are redacting those as well.

Reply:
Sure, I'll add that one to the test too!